packages/plugin-legacy/src/snippets.ts

```diff
@@ -9,7 +9,7 @@ export const systemJSInlineCode = `System.import(document.getElementById('${lega
 const detectModernBrowserVarName = '__vite_is_modern_browser'
 export const detectModernBrowserDetector =
   'import.meta.url;import("_").catch(()=>1);async function* g(){};'
-export const detectModernBrowserCode = `${detectModernBrowserDetector}window.${detectModernBrowserVarName}=true;`
+export const detectModernBrowserCode = `${detectModernBrowserDetector}if(window.location.protocol!="file:"){window.${detectModernBrowserVarName}=true}`
```

Comment:
```suggestion
export const detectModernBrowserCode = `${detectModernBrowserDetector}if(location.protocol!="file:"){window.${detectModernBrowserVarName}=true}`
```

We could make this a bit more compact. I checked the compat table and `location` vs `window.location` seems to have the same support.